esphome/core/helpers.h

```diff
@@ -651,35 +651,40 @@ void delay_microseconds_safe(uint32_t us);
 /// @name Memory management
 ///@{
 
-/** An STL allocator that uses SPI RAM.
+/** An STL allocator that uses SPI or internal RAM.
+ * Returns `nullptr` in case no memory is available.
  *
- * By setting flags, it can be configured to don't try main memory if SPI RAM is full or unavailable, and to return
- * `nulllptr` instead of aborting when no memory is available.
+ * By setting flags, it can be configured to:
+ * - perform external allocation falling back to main memory if SPI RAM is full or unavailable
+ * - perform external allocation only
+ * - perform internal allocation only
  */
-template<class T> class ExternalRAMAllocator {
+template<class T> class RAMAllocator {
  public:
   using value_type = T;
 
   enum Flags {
-    NONE = 0,
-    REFUSE_INTERNAL = 1 << 0,  ///< Refuse falling back to internal memory when external RAM is full or unavailable.
-    ALLOW_FAILURE = 1 << 1,    ///< Don't abort when memory allocation fails.
+    NONE =
+        0,  ///< Perform external allocation and fall back to internal memory when external RAM is full or unavailable.
+    ALLOC_EXTERNAL = 1 << 0,  ///< Perform external allocation only.
+    ALLOC_INTERNAL = 1 << 1,  ///< Perform internal allocation only.
+    ALLOW_FAILURE = 1 << 2,   ///< Does nothing. Kept for compatibility.
```

Comment:
Nitpicking: long comment breaks the line; comment indicators are unusual.

```suggestion
    NONE = 0,  // Perform external allocation and fall back to internal memory 
    ALLOC_EXTERNAL = 1 << 0,  // Perform external allocation only.
    ALLOC_INTERNAL = 1 << 1,  // Perform internal allocation only.
    ALLOW_FAILURE = 1 << 2,   // Does nothing. Kept for compatibility.
```

Reply:
Yup, this seemed strange for me, too, but I've kept the same format as the original comments thinking it was a sort of internal standard :)

Reply:
I have seen similar formatting in Espressif code, but it's not AFAIK commonly used in ESPHome and I don't personally seen any benefit in it.